### PR TITLE
doc: clarify when to use N-API

### PR DESCRIPTION
**Pull request by mhdawson**
Refs: https://github.com/nodejs/node-addon-api/issues/684

Signed-off-by: Michael Dawson <mdawson@devrus.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ the use of the C based [N-API](https://nodejs.org/dist/latest/docs/api/n-api.htm
...
 and exception handling semantics with low overhead.
 
+There are three options for implementing addons: N-API, nan, or direct
+use of internal V8, libuv and Node.js libraries. Unless there is a need for
+direct access to functionality which is not exposed by N-API as outline
+in [addons.md](https://github.com/nodejs/node/blob/master/doc/api/addons.md#c-addons)
+in Node.js core, use N-API. Refer to
+[C/C++ addons with N-API](https://github.com/nodejs/node/blob/master/doc/api/n-api.md)
+for more information on N-API.
+
```
Comment (NickNaso) [inline README.md:14]: What do you think to link the website doc instead?
- https://nodejs.org/dist/latest/docs/api/addons.html for C++ addons
- https://nodejs.org/dist/latest/docs/api/n-api.html for C++ addon with N-API

Comment (mhdawson) [reply README.md:14]: I had wondered about which would be best, happy to switch to website.

Comment (mhdawson) [reply README.md:14]: @NickNaso fixed.

**Review by legendecas — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -4,6 +4,14 @@ the use of the C based [N-API](https://nodejs.org/dist/latest/docs/api/n-api.htm
 provided by Node.js when using C++. It provides a C++ object model
 and exception handling semantics with low overhead.
 
+There are three options for implementing addons: N-API, nan, or direct
+use of internal V8, libuv and Node.js libraries. Unless there is a need for
+direct access to functionality which is not exposed by N-API as outline
+in [addons.md](https://nodejs.org/dist/latest/docs/api/addons.html)
```
[the remark this answers is not in the document]
Comment (mhdawson) [reply README.md]: @NickNaso updated

**Review by NickNaso — APPROVED** [review]
LGTM

**Pull request merged**